sdk/src/wallet/migration/migrate_3.rs

```diff
@@ -0,0 +1,1227 @@
+            }
+        }
+    }
+
+    #[derive(Deserialize)]
+    pub struct FoundryOutput {
+        pub amount: u64,
+        pub native_tokens: BoxedSlicePrefix<NativeToken>,
+        pub serial_number: u32,
+        pub token_scheme: TokenScheme,
+        pub unlock_conditions: BoxedSlicePrefix<UnlockCondition>,
+        pub features: BoxedSlicePrefix<Feature>,
+        pub immutable_features: BoxedSlicePrefix<Feature>,
+    }
+
+    #[derive(Serialize, Deserialize)]
+    #[serde(rename_all = "camelCase")]
+    pub struct FoundryOutputDto {
+        #[serde(rename = "type")]
+        pub kind: u8,
+        pub amount: String,
+        #[serde(skip_serializing_if = "Vec::is_empty", default)]
+        pub native_tokens: Vec<NativeToken>,
+        pub serial_number: u32,
+        pub token_scheme: TokenSchemeDto,
+        pub unlock_conditions: Vec<UnlockConditionDto>,
+        #[serde(skip_serializing_if = "Vec::is_empty", default)]
+        pub features: Vec<FeatureDto>,
+        #[serde(skip_serializing_if = "Vec::is_empty", default)]
+        pub immutable_features: Vec<FeatureDto>,
+    }
+
+    impl From<FoundryOutput> for FoundryOutputDto {
+        fn from(value: FoundryOutput) -> Self {
+            Self {
+                kind: 5,
+                amount: value.amount.to_string(),
+                native_tokens: value.native_tokens.inner,
+                serial_number: value.serial_number,
+                token_scheme: value.token_scheme.into(),
+                unlock_conditions: value.unlock_conditions.inner.into_iter().map(Into::into).collect(),
+                features: value.features.inner.into_iter().map(Into::into).collect(),
+                immutable_features: value.immutable_features.inner.into_iter().map(Into::into).collect(),
+            }
+        }
+    }
+
+    #[derive(Deserialize)]
+    pub struct NftOutput {
+        pub amount: u64,
+        pub native_tokens: BoxedSlicePrefix<NativeToken>,
+        pub nft_id: Value,
+        pub unlock_conditions: BoxedSlicePrefix<UnlockCondition>,
+        pub features: BoxedSlicePrefix<Feature>,
+        pub immutable_features: BoxedSlicePrefix<Feature>,
+    }
+
+    #[derive(Serialize, Deserialize)]
+    #[serde(rename_all = "camelCase")]
+    pub struct NftOutputDto {
+        #[serde(rename = "type")]
+        pub kind: u8,
+        pub amount: String,
+        #[serde(skip_serializing_if = "Vec::is_empty", default)]
+        pub native_tokens: Vec<NativeToken>,
+        pub nft_id: Value,
+        pub unlock_conditions: Vec<UnlockConditionDto>,
+        #[serde(skip_serializing_if = "Vec::is_empty", default)]
+        pub features: Vec<FeatureDto>,
+        #[serde(skip_serializing_if = "Vec::is_empty", default)]
+        pub immutable_features: Vec<FeatureDto>,
+    }
+
+    impl From<NftOutput> for NftOutputDto {
+        fn from(value: NftOutput) -> Self {
+            Self {
+                kind: 6,
```

Comment:
Any reason not to use the const here?
```suggestion
                kind: NftOutput::KIND,
```

Reply:
To make it self contained?

Reply:
No can do, can't use any outside definitions in case they change (even if you think it won't happen).